src/AutoRest.CSharp/Common/Output/Expressions/Samples/Snippets.ExampleValues.cs

```diff
@@ -390,13 +414,24 @@ private static ValueExpression GetExpressionForObjectType(ObjectType objectType,
                 foreach (var (property, exampleValue) in propertiesToWrite)
                 {
                     // we need to pass in the current type of this property to make sure its initialization is correct
-                    var propertyExpression = GetExpression(property.Declaration.Type, exampleValue, property.SerializationFormat, includeCollectionInitialization: false);
+                    var propertyExpression = GetExpression(property.Declaration.Type, exampleValue, GetSerializationFormat(property, property.Declaration.Type), includeCollectionInitialization: false);
                     initializerDict.Add(property.Declaration.Name, propertyExpression);
                 }
                 objectPropertyInitializer = new(initializerDict, false);
             }
 
             return new NewInstanceExpression(objectType.Type, arguments, objectPropertyInitializer);
+
+            static SerializationFormat GetSerializationFormat(ObjectTypeProperty property, CSharpType propertyType)
+            {
+                var serializationFormat = property.SerializationFormat; // this works for typespec input models
+                // TODO - wait for fix. For swagger input models, the serialization format is not properly added onto the property, therefore here we have to find the serialization format again
```

Comment:
issue opened for this here: https://github.com/Azure/autorest.csharp/issues/5158